string vendors
bool use_shortest_path
